android_app/app/src/main/java/com/health/openscale/core/OpenScale.java

```diff
@@ -385,6 +387,54 @@ public String getFilenameFromUri(Uri uri) {
         }
     }
 
+    public void importDatabase(Uri importFile) throws IOException {
+        File exportFile = context.getApplicationContext().getDatabasePath("openScale.db");
+        File tmpExportFile = context.getApplicationContext().getDatabasePath("openScale_tmp.db");
+
+        try {
+            copyFile(Uri.fromFile(exportFile), Uri.fromFile(tmpExportFile));
+            copyFile(importFile, Uri.fromFile(exportFile));
+
+            reopenDatabase();
+
+            if (!getScaleUserList().isEmpty()) {
+                selectScaleUser(getScaleUserList().get(0).getId());
+                updateScaleData();
+            }
+        } catch (SQLiteDatabaseCorruptException e) {
+            copyFile(Uri.fromFile(tmpExportFile), Uri.fromFile(exportFile));
+            throw new IOException(e.getMessage());
+        } finally {
+            tmpExportFile.delete();
+        }
+    }
+
+    public void exportDatabase(Uri exportFile) throws IOException {
+        File dbFile = context.getApplicationContext().getDatabasePath("openScale.db");
+
+        copyFile(Uri.fromFile(dbFile), exportFile);
+    }
+
+    private void copyFile(Uri src, Uri dst) throws IOException {
+        InputStream input = context.getContentResolver().openInputStream(src);
+        OutputStream output = context.getContentResolver().openOutputStream(dst);
+
+        try {
+            byte[] bytes = new byte[4096];
+            int count;
+
+            while ((count = input.read(bytes)) != -1){
+                output.write(bytes, 0, count);
+            }
+        } finally {
+            if (input != null)
+                input.close();
+            if (output != null)
+                output.flush();
+                output.close();
```

Comment:
Missing {} means this doesn't do what it looks like.

Reply:
true it is fixed.